### ISSUE TITLE
zc.buildout fails behind a cntlm proxy

### ISSUE THREAD

**Issue by ale-rt** [buildout/buildout#32]
When behind a cntlm proxy [1] used to authenticate to forefront [2], zc.buildout fails downloading files as reported in https://bugs.launchpad.net/zc.buildout/+bug/484735.
The patch in the bug report fixes the problem [3].

[1] http://cntlm.sourceforge.net
[2] http://technet.microsoft.com/en-us/forefront/ee807302
[3] the patch.
```diff
diff --git a/src/zc/buildout/download.py b/src/zc/buildout/download.py
index ee6865a..c0d5c3a 100644
--- a/src/zc/buildout/download.py
+++ b/src/zc/buildout/download.py
@@ -25,6 +25,7 @@ import re
 import shutil
 import tempfile
 import urllib
+import urllib2
 import urlparse
 import zc.buildout
 
@@ -177,7 +178,10 @@ class Download(object):
 
         try:
             try:
-                tmp_path, headers = urllib.urlretrieve(url, tmp_path)
+                tmp_sock = urllib2.urlopen(url)
+                tmp_file = open(tmp_path, 'w')
+                tmp_file.write(tmp_sock.read())
+                tmp_file.close()
                 if not check_md5sum(tmp_path, md5sum):
                     raise ChecksumError(
                         'MD5 checksum mismatch downloading %r' % url)
```

**Comment by stefano-m** [comment]
I have been bitten by the same issue on buildout 2.4.0 when trying to download and HTTPS url through an unauthenticated proxy.

This is likely related to the use of urllib's `urlretrieve` and `URLopener` on Python 2 that are affected by [Python issue 24599](https://bugs.python.org/issue24599), urllib2 on Python 2 does not suffer from the bug (fixed in [Python issue 1424152](https://bugs.python.org/issue1424152)).

I noticed that pull request #33 has been closed by @jimfulton and not merged. I think that a strategy to solve this problem that does not require to wait for the Python bug to be fixed and a new Python 2 release should be considered.

I know that his issue is quite old now, but I'd like to hear from the buildout developers before opening a new one.
